src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs

```diff
@@ -176,20 +175,27 @@ private void OnTimer(object sender, ElapsedEventArgs e)
                 return;
             }
 
-            DateTime now = DateTime.Now;
-            await InvokeJobFunction(now, false);
-
             // restart the timer with the next schedule occurrence
-            now = DateTime.Now;
-            DateTime nextOccurrence = _schedule.GetNextOccurrence(now);
-            TimeSpan nextInterval = nextOccurrence - now;
+            await InvokeJobFunction(DateTime.Now, false);
+            TimeSpan nextInterval = ScheduleStatus.Next - DateTime.Now;
             StartTimer(nextInterval);
         }
 
-        internal async Task InvokeJobFunction(DateTime lastOccurrence, bool isPastDue = false)
+        /// <summary>
+        /// Invokes the job function.
+        /// </summary>
+        /// <param name="invocationTime">The time of the invocation, likely DateTime.Now.</param>
+        /// <param name="isPastDue">True if the invocation is because the invocation is due to a past due timer.</param>
+        /// <param name="runOnStartup">True if the invocation is because the timer is configured to run on startup.</param>
+        internal async Task InvokeJobFunction(DateTime invocationTime, bool isPastDue = false, bool runOnStartup = false)
         {
             CancellationToken token = _cancellationTokenSource.Token;
-            TimerInfo timerInfo = new TimerInfo(_schedule, _scheduleStatus, isPastDue);
+            ScheduleStatus timerInfoStatus = null;
+            if (ScheduleMonitor != null)
+            {
+                timerInfoStatus = ScheduleStatus;
+            }
```

Comment:
If we're not doing schedule monitoring, then values like ScheduleStatus.Last will not be correct in your in memory case. So users coding against it in their function won't get valid values. So I'd say keep the back compat.